bidict/_common.py

```diff
@@ -41,53 +41,57 @@ def inv(self):
         return self._inv
 
     def __inverted__(self):
+        """
+        Returns an iterator over the inverse mappings.
+        """
         return iteritems(self._bwd)
 
     def __getitem__(self, key):
         return self._fwd[key]
 
-    def _put(self, key, val):
-        try:
-            oldval = self._fwd[key]
-        except KeyError:
-            oldval = _missing
-        try:
-            oldkey = self._bwd[val]
-        except KeyError:
-            oldkey = _missing
-
-        if oldval is not _missing and oldkey is not _missing:
-            if key == oldkey and val == oldval:
-                return
-            raise CollapseException((key, oldval), (oldkey, val))
-        elif oldval is not _missing:
-            del self._bwd[oldval]
-        elif oldkey is not _missing:
-            del self._fwd[oldkey]
-
+    def _put(self, key, val, overwrite_key=False, overwrite_val=True):
+        oldkey = self._bwd.get(val, _missing)
+        oldval = self._fwd.get(key, _missing)
+        if key == oldkey and val == oldval:
+            return
+        if oldval is not _missing:
+            if overwrite_val:
+                del self._bwd[oldval]
```

Comment:
Isn't there the potential here that this del will happen, and then the ValueExistsException gets raised in the other leg, leaving the map inconsistent?